### ISSUE TITLE
[bug] AssertionError: 'AvailableIpView' should either include aserializer_classattribute, or override the get_serializer_class() method

### ISSUE THREAD

**Issue by DavidMeu** [openwisp/openwisp-ipam#128]
While loading the docs endpoint:
`/api/v1/docs/`
Receiving the above error.

Full trace:
```
[04/Jan/2022 15:48:01] "GET /api/v1/docs/ HTTP/1.1" 200 2455 view's AvailableIpView raised exception during schema generation; usegetattr(self, 'swagger_fake_view', False)to detect and short-circuit this Traceback (most recent call last): File "\venv\lib\site-packages\drf_yasg\inspectors\base. py", line 42, in call_view_method return view_method() File "\venv\lib\site-packages\rest_framework\generics.p y", line 108, in get_serializer serializer_class = self.get_serializer_class() File "\venv\lib\site-packages\rest_framework\generics.py", line 122, in get_serializer_class assert self.serializer_class is not None, ( AssertionError: 'AvailableIpView' should either include a serializer_classattribute, or override theget_serializer_class()
method.
```

**Comment by pandafy** [comment]
There are some views in [openwisp_ipam/api/views.py](https://github.com/openwisp/openwisp-ipam/blob/master/openwisp_ipam/api/views.py) that inherits from API view classes of DRF, but they are not utilising features of DRF. `AvailableIpView` is one of such views. I think we should update these views to Inherit from `django.views.generics.View` instead.

**Comment by pandafy** [comment]
Alternatively, we can use the base SerializerClass in such API views. 

```diff
diff --git a/openwisp_ipam/api/views.py b/openwisp_ipam/api/views.py
index f3a5e10..725c16c 100644
--- a/openwisp_ipam/api/views.py
+++ b/openwisp_ipam/api/views.py
@@ -157,6 +157,7 @@ class HostsSet:
 class AvailableIpView(ProtectedAPIMixin, IpAddressOrgMixin, RetrieveAPIView):
     subnet_model = Subnet
     queryset = IpAddress.objects.none()
+    serializer_class = serializers.Serializer
 
     def get(self, request, *args, **kwargs):
         subnet = get_object_or_404(self.subnet_model, pk=self.kwargs['subnet_id'])
```

**Comment by devkapilbansal** [comment]
> Alternatively, we can use the base SerializerClass in such API views.

If we aren't utilising GenericViews, then I don't see any point in using them. Shouldn't they be replaced by `rest_framework.views.ApiView` instead?

**Comment by pandafy** [comment]
> If we aren't utilising GenericViews, then I don't see any point in using them. Shouldn't they be replaced by `rest_framework.views.ApiView` instead?

I tried doing  this locally, and it didn't work 
```
Internal Server Error: /api/v1/ipam/subnet/08058465-0925-4b52-876c-238ddc90d3d3/get-next-available-ip/
Traceback (most recent call last):
  File "/home/pandafy/openwisp/venv-ipam/lib/python3.8/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/home/pandafy/openwisp/venv-ipam/lib/python3.8/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/pandafy/openwisp/venv-ipam/lib/python3.8/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "/home/pandafy/openwisp/venv-ipam/lib/python3.8/site-packages/django/views/generic/base.py", line 69, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/pandafy/openwisp/venv-ipam/lib/python3.8/site-packages/rest_framework/views.py", line 509, in dispatch
    response = self.handle_exception(exc)
  File "/home/pandafy/openwisp/venv-ipam/lib/python3.8/site-packages/rest_framework/views.py", line 469, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/home/pandafy/openwisp/venv-ipam/lib/python3.8/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
    raise exc
  File "/home/pandafy/openwisp/venv-ipam/lib/python3.8/site-packages/rest_framework/views.py", line 497, in dispatch
    self.initial(request, *args, **kwargs)
  File "/home/pandafy/openwisp/venv-ipam/lib/python3.8/site-packages/rest_framework/views.py", line 415, in initial
    self.check_permissions(request)
  File "/home/pandafy/openwisp/venv-ipam/lib/python3.8/site-packages/rest_framework/views.py", line 332, in check_permissions
    if not permission.has_permission(request, self):
  File "/home/pandafy/openwisp/venv-ipam/lib/python3.8/site-packages/rest_framework/permissions.py", line 229, in has_permission
    queryset = self._queryset(view)
  File "/home/pandafy/openwisp/venv-ipam/lib/python3.8/site-packages/rest_framework/permissions.py", line 212, in _queryset
    queryset = view.get_queryset()
  File "/home/pandafy/openwisp/venv-ipam/lib/python3.8/site-packages/openwisp_users/api/mixins.py", line 77, in get_queryset
    qs = super().get_queryset()
AttributeError: 'super' object has no attribute 'get_queryset'
[27/Jan/2022 11:26:33] "GET /api/v1/ipam/subnet/08058465-0925-4b52-876c-238ddc90d3d3/get-next-available-ip/ HTTP/1.1" 500 115228
```
